docker/docker.go

```diff
@@ -42,27 +43,117 @@ func CreateNetwork(l zerolog.Logger) (*tc.DockerNetwork, error) {
+type StartContainerRetrier func(l zerolog.Logger, startErr error, req tc.GenericContainerRequest) (tc.Container, error)
+
+var NaiveRetrier = func(l zerolog.Logger, startErr error, req tc.GenericContainerRequest) (tc.Container, error) {
+	l.Debug().
+		Str("Start error", startErr.Error()).
+		Str("Retrier", "NaiveRetrier").
+		Msgf("Attempting to start %s container", req.Name)
+
+	ct, err := tc.GenericContainer(testcontext.Get(nil), req)
+	if err == nil {
+		l.Debug().
+			Str("Retrier", "NaiveRetrier").
+			Msgf("Successfully started %s container", req.Name)
+		return ct, nil
+	}
+	if ct != nil {
+		err := ct.Terminate(testcontext.Get(nil))
+		if err != nil {
+			l.Error().
+				Err(err).
+				Msgf("Cannot terminate %s container to initiate restart", req.Name)
+			return nil, err
+		}
+	}
+	req.Reuse = false
+
+	l.Debug().
+		Str("Original start error", startErr.Error()).
+		Str("Current start error", err.Error()).
+		Str("Retrier", "NaiveRetrier").
+		Msgf("Failed to start %s container,", req.Name)
+
+	return nil, startErr
+}
+
+var LinuxPlatoformImageRetrier = func(l zerolog.Logger, startErr error, req tc.GenericContainerRequest) (tc.Container, error) {
+	// if it's nil we don't know if we can handle it so we won't try
+	if startErr == nil {
+		return nil, startErr
+	}
+
+	// a bit lame, but that's the lame error we get in case there's no specific image for our platform :facepalm:
+	if !strings.Contains(startErr.Error(), "No such image") {
+		l.Debug().
+			Str("Start error", startErr.Error()).
+			Str("Retrier", "PlatoformImageRetrier").
+			Msgf("Won't try to start %s container again, unsupported error", req.Name)
+		return nil, startErr
+	}
+
+	l.Debug().
+		Str("Start error", startErr.Error()).
+		Str("Retrier", "PlatoformImageRetrier").
+		Msgf("Attempting to start %s container", req.Name)
+
+	originalPlatform := req.ImagePlatform
+	req.ImagePlatform = "linux/x86_64"
+
+	ct, err := tc.GenericContainer(testcontext.Get(nil), req)
+	if err == nil {
+		l.Debug().
+			Str("Retrier", "PlatoformImageRetrier").
+			Msgf("Successfully started %s container", req.Name)
+		return ct, nil
+	}
+
+	req.ImagePlatform = originalPlatform
+
+	if ct != nil {
+		err := ct.Terminate(testcontext.Get(nil))
+		if err != nil {
+			l.Error().Err(err).Msgf("Cannot terminate %s container to initiate restart", req.Name)
+			return nil, err
+		}
+	}
+
+	l.Debug().
+		Str("Original start error", startErr.Error()).
+		Str("Current start error", err.Error()).
+		Str("Retrier", "PlatoformImageRetrier").
+		Msgf("Failed to start %s container,", req.Name)
+
+	return nil, startErr
+}
+
+// StartContainerWithRetry attempts to start a container with 3 retry attempts.
+// It will try to start the container with the provided retriers, if none are provided it will use the default retriers.
+// Default being: 1. tries to download image for "linux/x86_64" platform 2. simply starts again without changing anything
```

Comment:
@Tofel what if there is an image for apple silicon? Shouldn't it be downloaded first? I wonder if you could use `platform := runtime.GOOS + "/" + runtime.GOARCH` to determine platform and first try to download native image. And, if it does not exist, fallback to any image.

Reply:
ah, I guess `NaiveRetrier` tries to download the native image? If yes, maybe but it first on the list in line [144](https://github.com/smartcontractkit/chainlink-testing-framework/pull/817/files#diff-3b425d7dadd18dc6d2fadc2bc169afae1f7bbc8f757a66d2837e9e78fb792c90R144)

Reply:
but it tries to download image for your current platform by default, and then this retry shouldn't even be triggered, no?

Reply:
ok, I thought we first use `Linux` `retriers = append(retriers, LinuxPlatoformImageRetrier, NaiveRetrier)`. But, these are retries and the first attempt is current platform (`ct, err = tc.GenericContainer(testcontext.Get(nil), req)`?

Reply:
so that's how it works:
* first attempt is just whatever you passed in your container request
* if that fails, we go for retries
* first retry is with linux image platform (maybe here I could also check if the platform isn't already set to it, and if it is, skip that retry)
* then we try the naive retry, which is what we had before, which is just trying to start again without modifying anything

Reply:
great, thanks!